packages/theme-sections/section.js

```diff
@@ -0,0 +1,140 @@
+var SECTION_ID_ATTR = 'data-section-id';
+
+export default function Section(container, properties) {
+  this.container = validateContainerElement(container);
+  this.id = container.getAttribute(SECTION_ID_ATTR);
+  this.extensions = [];
+
+  assign(this, validatePropertiesObject(properties));
+
+  this._onUnload = this._onUnload.bind(this);
+  this._onSelect = this._onSelect.bind(this);
+  this._onDeselect = this._onDeselect.bind(this);
+  this._onBlockSelect = this._onBlockSelect.bind(this);
+  this._onBlockDeselect = this._onBlockDeselect.bind(this);
+
+  document.addEventListener('shopify:section:unload', this._onUnload);
+  document.addEventListener('shopify:section:select', this._onSelect);
+  document.addEventListener('shopify:section:deselect', this._onDeselect);
+  document.addEventListener('shopify:block:select', this._onBlockSelect);
+  document.addEventListener('shopify:block:deselect', this._onBlockDeselect);
+
+  this.onLoad();
+}
+
+Section.prototype = {
+  onLoad: function onLoad() {},
```

Comment:
It sets `onLoad` to an empty function, so that we can reliably always call it without checking if the user included an `onLoad` method when they registered a section with `section.register()`